plugins/scaffolder-backend-module-gitlab/src/actions/gitlabMergeRequest.ts

```diff
@@ -27,6 +28,20 @@ import { resolveSafeChildPath } from '@backstage/backend-plugin-api';
 import { createGitlabApi } from './helpers';
 import { examples } from './gitlabMergeRequest.examples';
 
+function getFileAction(
+  file: SerializedFile,
+  remoteFiles: Types.RepositoryTreeSchema[],
+  defaultCommitAction: 'create' | 'delete' | 'update' | 'auto' | undefined,
+): 'create' | 'delete' | 'update' {
+  if (defaultCommitAction === 'auto') {
+    return remoteFiles &&
+      remoteFiles.some(remoteFile => remoteFile.path === file.path)
+      ? 'update'
+      : 'create';
+  }
+  return defaultCommitAction ?? 'create';
```

Comment:
It would be better to have the default action as 'auto' instead of 'create'

Reply:
Yep ! Why not